src/components/ImageUpload/ImageUploadProvider.tsx

```diff
@@ -0,0 +1,279 @@
+
+      results
+        .then(() => {
+          if (uploadQueue.current.size === 0) {
+            // All uploads finished
+            clearSnackbar();
+            showErrorSnackbars(
+              showSnackbar,
+              uploadErrors.current,
+              t,
+              maxUploadSizeBytes,
+            );
+            uploadErrors.current = { size: [], resolution: [], other: [] };
+          }
+          return;
+        })
+        .catch(() => {
+          // ignore
+        });
+
+      return results;
+    },
+    [
+      clearSnackbar,
+      handleRejectedFile,
+      maxUploadSizeBytes,
+      showSnackbar,
+      uploadImage,
+      t,
+    ],
+  );
+
+  return (
+    <ImageUploadContext.Provider value={{ handleDrop, maxUploadSizeBytes }}>
+      {children}
+    </ImageUploadContext.Provider>
+  );
+}
+
+/**
+ * Show snackbars for the different types of errors.
+ *
+ * @param showSnackbar - Function to show a snackbar {@link SnackbarState.showSnackbar}
+ * @param errors - Errors for which a snackbar should be displayed
+ * @param t - Translation function
+ * @param maxUploadSizeBytes - Maximum allowed upload size in bytes
+ */
+function showErrorSnackbars(
+  showSnackbar: (snackbar: SnackbarProps) => void,
+  errors: UploadErrors,
+  t: TFunction,
+  maxUploadSizeBytes: number,
+): void {
+  if (errors.size.length > 0) {
+    const message = t(
+      'imageUpload.sizeError',
+      'The file {{fileNames}} cannot be uploaded due to its size. The upload is possible for a maximum of {{limit}} per file.',
+      {
+        count: errors.size.length,
+        fileNames: errors.size.join(', '),
+        limit:
+          (maxUploadSizeBytes / 1048576).toLocaleString(undefined, {
+            maximumFractionDigits: 2,
+          }) + ' MiB',
+      },
+    );
+
+    showSnackbar({
+      key: message,
+      message,
+      action: <SnackbarDismissAction />,
+      autoHideDuration: 10000,
+    });
+  }
+
+  if (errors.resolution.length > 0) {
+    const message = t(
+      'imageUpload.resolutionError',
+      'The file {{fileNames}} cannot be uploaded due to its resolution. The upload is possible for a maximum of {{limit}} per file.',
+      {
+        count: errors.size.length,
+        fileNames: errors.resolution.join(', '),
+        limit:
+          (maxResolutionPixels / 1_000_000).toLocaleString(undefined, {
+            maximumFractionDigits: 2,
+          }) + ' MP',
+      },
+    );
+
+    showSnackbar({
+      key: message,
+      message,
+      action: <SnackbarDismissAction />,
+      autoHideDuration: 10000,
+    });
+  }
+
+  if (errors.other.length > 0) {
+    const message = t(
+      'imageUpload.otherError',
+      'The file {{fileNames}} cannot be uploaded.',
+      {
+        count: errors.other.length,
+        fileNames: errors.other.join(', '),
+      },
+    );
+
+    showSnackbar({
+      key: message,
+      message,
+      action: <SnackbarDismissAction />,
+      // autoHideDuration: 10000,
```

Comment:
Please delete commented out code if not needed.

Reply:
Commented in again. Also added tests to check that.